src/adaptor/discord/member-stats.ts

```diff
@@ -42,4 +46,26 @@ export class DiscordMemberStats
       roles: roles.cache.size
     }));
   }
+
+  async fetchStats(userId: string): Promise<UserStats | null> {
+    const guild = await this.client.guilds.fetch(this.guildId);
+    const member = await guild.members.fetch(userId);
+    if (!member) {
+      return null;
+    }
+
+    const joinedAt: Date | undefined = member.joinedAt ?? undefined;
+    const hoistRoleId = (member.roles.hoist?.id as Snowflake) ?? null;
```

Comment:
```suggestion
    const hoistRoleId = (member.roles.hoist?.id as Snowflake) ?? undefined;
```
`Snowflake | undefined` の型になっていたのでこうするべきかと